docs/src/transactions/transaction-execution.md

```diff
@@ -0,0 +1,52 @@
+- the `BlockHeader`, which contains metadata about the block, commitments to the current state of the chain and the hash of the proof that attests to the integrity of the chain.
+- the `ChainMmr`, which allows for efficient authentication of consumed notes during transaction execution. Authentication is achieved by providing an inclusion proof for the consumed notes in the transaction against the `ChainMmr`-root associated with the latest block known at the time of transaction execution.  
+- the `InputNotes` that are being consumed in the transaction (InputNotes), including the corresponding note data, e.g. the [note script](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#script) and [serial number](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#serial-number).
+
+_Note: The `InputNotes` must all be already recorded on-chain in order for the transaction to succeed._
+
+## The Transaction Compiler
+Every transaction must be executed within the Miden VM to generate a transaction proof. In Miden there is a proof for every transaction. The transaction compiler is responsible for building executable programs. The generated programs - MASM programs - can then be executed on the Miden VM which generates a zkProof. In addition to transaction compilation, the transaction compiler provides methods which can be used to compile Miden account code, note scripts, and transaction scripts. 
+
+The `compile_transaction()` function results in an executable MASM Program, including the provided account interface and notes, an optional transaction script and the [Transaction Kernel Program](transaction-kernel.md). The Transaction Kernel Program defines procedures and the memory layout for all parts of the transaction. A detailed description can be found in the next section. 
+
+Finally, after the transaction program has been compiled and the inputs including the advice provider were correctly populated, the transaction can be executed.
+
+## The Executed Transaction and the Transaction Outputs
+The `ExecutedTransaction` object represents the result of a transaction - not its proof yet. From it, the account, and storage delta can be extracted. Furthermore, it serves as an input of the transaction prover to generate the proof. A successfully executed transaction results in a new state of the provided account, a vector of all created Notes (`OutputNotes`) and a vector of all the consumed Notes (`InputNotes`) and their Nullifiers.
+
+```
+pub struct ExecutedTransaction {
+    id: OnceCell<TransactionId>,
+    program: Program,
+    tx_inputs: TransactionInputs,
+    tx_outputs: TransactionOutputs,
+    account_delta: AccountDelta,
+    tx_script: Option<TransactionScript>,
+    advice_witness: AdviceInputs,
+}
+```
```

Comment:
I would not use Rust structs to illustrate these. Things like `id: OnceCell<TransactionId>` are internal implementation detail and probably will confuse people more than help.

A general comment for this section: I think we should make these docs as independent from the code as possible. These are describing concepts and how they come together. Naming of functions and structs is probably too much detail here.

Reply:
I removed the Rust code and rewrote the chapter.

Maybe you are right and we should not explain code. But shouldn't we somewhere explain it? At least, we need to be consistent. We explain the MASM procedures in detail in the next section.